pyvista/core/filters/image_data.py

```diff
@@ -2287,3 +2287,31 @@ def _validate_dimensional_operation(
             )
 
         return dimensions_mask, dimensions_result
+
+    def magnify(
+        self, *, factor: VectorLike[int] = (1, 1, 1), progress_bar: bool = False
+    ) -> pyvista.ImageData:
+        """Magnify an image by an integer value.
+
+        Parameters
+        ----------
+        factor : VectorLike[int], optional
+           The integer magnification factors in the i-j-k directions.
+           Initially, factors are set to 1 in all directions.
+
+        progress_bar : bool, default: False
+           Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.ImageData
+           Magnified image.
+
+        """
+        _validation.validate_array(factor, must_have_shape=3, must_be_integer=True, name='factor')
+        _validation.check_instance(progress_bar, bool)
+        alg = _vtk.vtkImageMagnify()
+        alg.SetInputDataObject(self)
+        alg.SetMagnificationFactors(factor)
```

Comment:
With `VectorLike` typed inputs I have found that unpacking the variable is the easiest way to resolve type errors where a more narrow `Sequence` typed input is expected by VTK, since these methods are usually overloaded such that both `(factor: Sequence[float])` and `(factor_x: float, factor_y: float, factor_z: float)` are allowed type-wise.

```suggestion
        alg.SetMagnificationFactors(*factor)
```
This should fix the currently reported `mypy` error.